R/Utils.R

```diff
@@ -196,4 +198,74 @@ SeuratToAnnData <- function(seuratObj, outFileBaseName, assayName = NULL, doDiet
   }
 
   return(ret)
+}
+
+#' @title Remove UCell Doublets
+#'
+#' @description Attempts to remove dump-gate "doublets" from the main three cell type subset objects (Tcell, Bcell, MoMacDC)
+#' @param parentUCell The subset's target cell type's UCell score. (Computed by scGate early in processing).
+#' @param otherUCellScores A vector of cell type UCell scores to attempt to discard from downstream processing. By default, these are cell types that comprise the dump gate in RIRA's scGate model. 
+#' @param clusterResolution The metadata column name that defines cluster membership (e.g. ClusterNames_0.2)
+#' @param cosineSimilarityCutoff A number between -1 and 1 that is used to threshold/gate the prospective doublets as defined by the computed score.
+#' @return Returns a seurat object with the UCell doublets removed. 
+#' @export
+#' 
+.RemoveUCellDoublets <- function(seuratObj, parentUCell = NULL, otherUCellScores = c("Platelet_MK_UCell", "Erythrocyte_UCell", "Neutrophil_UCell", "Epithelial_UCell", "Stromal_UCell"), clusterResolution = "ClusterNames_1.2", cosineSimilarityCutoff = 0.5){
+  #test for valid parentUCell
+  if(is.null(parentUCell)){
+    stop("Please define a parent population's UCell score column within the seurat object's metadata. For RIRA, this should be one of: 'Tcell_UCell', 'Bcell_UCell', or 'MoMacDC_UCell'.")
```

Comment:
Be aware, our scGate is actually based on the labels like Tcell.RM, Bcell.RM, etc.

Reply:
Got it - so the UCell scores of importance are something like: Tcell_RM_Ucell or Tcell.RM_Ucell?

Reply:
Importance is subjective, but the ones used for classifying rhesus cells are here: https://github.com/bimberlabinternal/RIRA/blob/25a5e732cbfe5ddb5146ac4df9c62be3dc8903ee/R/scGate.R#L165. 

I know this is built by using a RBC/Platelet UCell, but if there a rationale to comparing that score to the celltypist probability, since that can score a more complete set of cells than the more minimal scGate/UCell score?